datafusion/sql/src/planner.rs

```diff
@@ -1737,7 +1737,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             SQLExpr::CompoundIdentifier(ids) => {
                 let mut var_names: Vec<_> = ids.into_iter().map(|s| normalize_ident(&s)).collect();
 
-                if &var_names[0][0..1] == "@" {
+                if &var_names[0].get(0..1) == &Some("@") {
```

Comment:
I am not sure it is possible to get an empty variable name here from the sqlparser

`CompoundIdentifiers` are things like `foo.bar`. I tried the following in datafusion-cli and could not provoke the issue:

```sql
❯ 
select foo.@f from foo;
SchemaError(FieldNotFound { qualifier: Some("foo"), name: "@f", valid_fields: Some(["foo.d"]) })
❯ 
select foo.@ from foo;
SchemaError(FieldNotFound { qualifier: Some("foo"), name: "@", valid_fields: Some(["foo.d"]) })
❯ 
select foo. from foo;
SchemaError(FieldNotFound { qualifier: Some("foo"), name: "from", valid_fields: Some([]) })
❯ 
select foo.d.foo from foo;
NotImplemented("Unsupported compound identifier '[\"foo\"]'")
❯ 
select foo.@.foo from foo;
NotImplemented("Unsupported compound identifier '[\"foo\"]'")
```

Thus I don't think a test is possible (and there not needed). Thank you for the change anyways